### PR TITLE
feat(string): add trimStart #808

### LINKED ISSUE #808: trimStart/trimEnd
https://github.com/tc39/proposal-string-left-right-trim

Don't forget appropriate polyfills + aliases (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart)

### PR DESCRIPTION
**Pull request by rafaelpadovezi**
Closes #808

### CODE REVIEW

File: src/trimStart.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import polyfill from './internal/polyfills/String.trimStart';
+import isFunction from './isFunction';
+
+export const trimStartPolyfill = polyfill;
+
+export const trimStartInvoker = invoker(0, 'trimStart');
+
+/**
+ * Removes whitespace from the beginning of a string
```
Comment (char0n) [inline src/trimStart.js]: string -> string. (dot at the end)

File: src/trimStart.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { invoker } from 'ramda';
+
+import polyfill from './internal/polyfills/String.trimStart';
+import isFunction from './isFunction';
+
+export const trimStartPolyfill = polyfill;
+
+export const trimStartInvoker = invoker(0, 'trimStart');
```
Comment (char0n) [inline src/trimStart.js]: No need to export invoker here

File: test/trimStart.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      });
+    });
+
+    context('given an string with whitespaces only', function() {
+      specify('should return an empty string', function() {
+        eq(trimStartPolyfill('   '), '');
+      });
+    });
+  });
+});
```
Comment (char0n) [inline test/trimStart.js]: We're missings tests in for currying in every top level context

File: types/test/trimStart.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import * as RA from 'ramda-adjunct';
+
+RA.trimStart('abc'); // $ExpectType string
```
Comment (char0n) [inline types/test/trimStart.ts]: Missing tests for `trimLeft` here. We can put them in this file directly.

File: src/trimStart.js (hunk 2 above)
Comment (char0n) [reply src/trimStart.js]: Taking this back. This is consistent with other part of the codebase (replaceAll)

**Comment by rafaelpadovezi** [comment]
Thanks for the comments!

**Comment by char0n** [comment]
Thanks for contributing !

**Pull request merged**